### PR TITLE
Cross-scheduler inconsistent result structure

### LINKED ISSUE #8474: Cross-scheduler inconsistent result structure
**What happened**:

Due to a mistake in our code, we were persisting a dask dataframe in one scheduler, but then ran the compute while specifying threads scheduler. What was weird was that the computation returned a pandas Series of Futures instead of the expected DataFrame (or exception).

**What you expected to happen**:

I'm not sure what the best behavior is. Could be erroring on conflicting scheduler, running on the original scheduler, or running on the specified scheduler. Either way, the result structure should be consistent.

**Minimal Complete Verifiable Example**:

```python
from distributed import Client, LocalCluster
import pandas as pd
import dask.dataframe as dd

cluster = LocalCluster()
client=Client(cluster)

df = pd.DataFrame({'a': [1,2,3]})

ddf = dd.from_pandas(df, npartitions=3)

ddf = client.persist(ddf)

# Works, and returns a DF as expected
print(ddf.compute())

# Returns a pandas Series with a row per-partition with a future as the result
print(ddf.compute(scheduler="threads"))
```

**Anything else we need to know?**:

**Environment**:

- Dask version:
- Python version:
- Operating System:
- Install method (conda, pip, source):

### PR DESCRIPTION
**Pull request by ppsbs**
Raises a warning when `compute` or `persist` are called with a scheduler different from "dask.distributed" or "distributed" in Dask.distributed mode

- [X] Closes #8474
- [X] Tests added / passed
- [X] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by ppsbs** [comment]
Basically I check (inside of `get_scheduler`) for definitions of local schedulers in a distributed environment. If that's the case I raise a warning. Unfortunately, a few tests in `test_distributed` and `test_hdf` were doing that through `assert_eq` calls, so I had to modify them so they could ignore those warnings.

File: dask/tests/test_distributed.py [hunk 1]
```diff
@@ -130,8 +130,20 @@ def test_futures_to_delayed_array(c):
     assert_eq(A.compute(), np.concatenate([x, x], axis=0))
 
 
+# with pytest.warns(UserWarning, match=warning_message) as user_warnings_b:
+#     get_scheduler(scheduler="dask.distributed")
+#     get_scheduler(scheduler="distributed")
+#     get_scheduler()
+# assert len(user_warnings_b) == 0
```
Comment (phobson) [inline dask/tests/test_distributed.py:137]: Was this intended to be added as a proper test?

Comment (ppsbs) [reply dask/tests/test_distributed.py]: I'm sorry, I forgot to remove those comments. I just pushed a new version after removing them.

**Comment by scharlottej13** [comment]
@jrbourbeau would you be able to take a look at this?

File: dask/base.py [hunk 2]
```diff
@@ -1250,7 +1251,13 @@ def get_scheduler(get=None, scheduler=None, collections=None, cls=None):
             return scheduler.get
         elif isinstance(scheduler, str):
             scheduler = scheduler.lower()
+
             if scheduler in named_schedulers:
+                if config.get("scheduler", None) in ("dask.distributed", "distributed"):
+                    warnings.warn(
+                        "Passing a local execution scheduler in Dask.distributed might "
+                        "lead to unexpected results."
```
Comment (jsignell) [inline dask/base.py:1259]: I think this wording is potentially confusing. How about:

```suggestion
                        "Running on a single-machine scheduler when a distributed client "
                        "is active might lead to unexpected results."
```

File: dask/tests/test_distributed.py [hunk 3]
```diff
@@ -293,6 +302,10 @@ def reliable_double(x):
         "full",
     ],
 )
+@pytest.mark.filterwarnings(
+    "ignore:Passing a local execution scheduler in Dask.distributed might "
+    "lead to unexpected results."
+)
```
Comment (jsignell) [inline dask/tests/test_distributed.py:308]: I don't see where the scheduler is getting set explicitly in this test.

**Review by jsignell — COMMENTED** [review]
Thanks for working on this @ppsbs - I just have a few comments

File: dask/base.py (hunk 2 above)
Comment (ppsbs) [reply dask/base.py:1259]: Hi @jsignell , I think your wording is better and I will updated it in my next PR.

File: dask/tests/test_distributed.py (hunk 3 above)
Comment (jsignell) [reply dask/tests/test_distributed.py:308]: Oof I was going to say maybe we can set the scheduler in the `assert_eq` call instead, but that does not work very well. The `"sync"` scheduler is still hardcoded in some of the nested functions and if you switch them to take scheduler as an input then you run into other issues. So I think this is fine as you have it for now, and we can open a separate ticket to make the `assert_eq` work better with distritbuted client.

**Comment by jsignell** [comment]
ok to test

File: dask/tests/test_distributed.py (hunk 3 above)
Comment (jsignell) [reply dask/tests/test_distributed.py:308]: Oh! I forgot to comment. I merged a change in https://github.com/dask/dask/pull/8755 that fixes the hard coding. So I think you can get rid of some of these warning filters. 

> I see that you triggered the tests. In this case, should I still update the wording of the warning message as you suggested? Thanks!

Yes please update the wording

Comment (ppsbs) [reply dask/tests/test_distributed.py]: Cool, just pushed the fixes. Btw, do you know if there's a more elegant way to deal with the warning messages? Right now we need to "hard-code" the messages everywhere we need it (in the filters, for example). I'm just curious about it!

**Review by jsignell — APPROVED** [review]
This looks good to me! Thanks for sticking with this @ppsbs. With regards to the error message. I am actually pretty happy with having such an explicit filter even if it is verbose. It will help keep these tests maintainable because we know exactly what we are trying to filter.

**Pull request merged**